### ISSUE TITLE
Can acceptedName distinguish between foldername and filename?

### ISSUE THREAD

**Issue by donShakespeare** [Studio-42/elFinder#1959]
In conjunction with mimeTypes, for added security, I have a function that forces acceptedName to not start with dot, and to force an extension that is predefined in an array.

Folders don't have extension, therefore this function will prevent folders from being created or renamed. This is undesirable.
Anybody have ideas?
Thanks.

**Comment by nao-pon** [comment]
@donShakespeare OK. I'll make a new configuration option `acceptedDirname` to control the directory name.

Thanks! 👍

**Comment by nao-pon** [comment]
Done. @donShakespeare you can use `acceptedDirname` in volume roots options with [nightly build](https://github.com/Studio-42/elFinder/archive/2.1.zip) or elFinder >= 2.1.24.

Thanks!

**Comment by donShakespeare** [comment]
Excellent! Thank you. Makes a world of difference now.

1. Will this remain as accepted**Dirn**ame or accepted**DirN**ame

2. Will there be corresponding changes in the JS ui to return `Invalid folder name` instead of `Invalid file name`?

3. Will the wonderful `Sanitizer` Plugin also have this feature?
```
private function sanitizeFolderName($filename, $opts, $allows = array()) {
  $targets = $allows? array_diff($opts['targets'], $allows) : $opts['targets'];
  return str_replace($targets, $opts['replace'], $filename);
}
```
or at atleast
```
private function sanitizeFileName($filename, $origin, $opts, $allows = array()) {
  if($origin != "folder"){
    $targets = $allows? array_diff($opts['targets'], $allows) : $opts['targets'];
    return str_replace($targets, $opts['replace'], $filename);
  }
}
```

**Comment by nao-pon** [comment]
Thanks for the suggestion!

> Will this remain as acceptedDirname or acceptedDirName

Yes, It is determined as `acceptedDirname`.

> Will there be corresponding changes in the JS ui to return Invalid folder name instead of Invalid file name?

Yes, I'll implement it.

> Will the wonderful Sanitizer Plugin also have this feature?

Yes, I'll add an option `pathAllows` for to make configurable to characters allowed in path name.

**Comment by nao-pon** [comment]
Done it. Thanks!

**Comment by donShakespeare** [comment]
Works beautifully!

I noticed this code `errInvDirame` not translating; is that something that will be fixed when the release is made? (Invalid folder name)
```
Unable to rename "myFolder".
errInvDirame

**Comment by nao-pon** [comment]
> I noticed this code errInvDirame not translating; is that something that will be fixed when the release is made? (Invalid folder name)

Yes, it will be included in 2.1.24. Until then you can try it with the [nightly build](https://github.com/Studio-42/elFinder/archive/2.1.zip). When you try, do not forget to clear the browser cache.
